In [3]:
# default_exp engineering.docker

%reload_ext autoreload
%autoreload 2

https://www.jianshu.com/p/82ef099d5925

# images镜像相关命令

## 查看本地已经加载的image

In [3]:
!docker images

REPOSITORY                                      TAG                 IMAGE ID            CREATED             SIZE
hub-dev.leapstack.cn/models/rc_scores_predict   v1.4                65f0ffecf7ce        41 minutes ago      1.49GB
hub-dev.leapstack.cn/models/rc_scores_predict   v1.3                547bdfde9011        4 days ago          1.49GB
hub-dev.leapstack.cn/models/rc_scores_predict   v1.2                95bc1a6d6776        13 days ago         1.49GB
<none>                                          <none>              07717971964c        3 weeks ago         1.49GB
<none>                                          <none>              5f3f9e305b20        4 weeks ago         1.48GB
hub-dev.leapstack.cn/models/rc_scores_predict   v1.0                49f0a5ad72dd        4 weeks ago         1.48GB
<none>                                          <none>              26bc6f4ebcb5        4 weeks ago         1.51GB
harbor.leapstack.cn/base/python36               predict-base-v1     e291fb1f1064  

## 删除image

In [2]:
!docker rmi 925f9bae639d

Untagged: hub-dev.leapstack.cn/models/rc_scores_predict:v1.1
Deleted: sha256:925f9bae639d1e0b4ed4cae47d61c4622764ba6d268105a5844a5382a431e8b6
Deleted: sha256:cfa2246d86571bc9cd0ef48a30337c97f8169f001d8a11802557a93e9eecc64a


## 把image输出为本地镜像文件

In [ ]:
docker save hub-dev.leapstack.cn/models/rc_scores_predict:v1.4 -o rc_scores_predict.img

## 把本地image文件装载进docker

In [ ]:
docker load -i rc_scores_predict.img

## 把容器提交为镜像

In [2]:
!docker commit -m="提交的描述信息" -a="作者" 容器ID a/b:v1.2

Object `commit` not found.


In [ ]:
!docker commit

In [ ]:
!docker commit

# 容器container相关命令

## 查看容器
docker ps：查看当前正在运行的容器

docker ps -a：查看所有容器

In [4]:
!docker ps

CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES


In [5]:
!docker ps -a

CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                     PORTS                              NAMES
d768cab2bb10        26bc6f4ebcb5        "container-entrypoin…"   4 weeks ago         Exited (255) 4 weeks ago   8080/tcp, 0.0.0.0:9092->9092/tcp   stoic_euler
ba5fbe22a1f5        26bc6f4ebcb5        "container-entrypoin…"   4 weeks ago         Created                                                       clever_austin


## 启动容器 docker run
运行这个命令相当于执行了容器生命周期的create & start操作

    docker run -it java java -version

-i 	以交互模式运行容器，通常与 -t 同时使用；
-t 	为容器重新分配一个伪输入终端，通常与 -i 同时使用；
-d 	后台运行容器，并返回容器ID；


--name containername ：给容器命名
第一个java：image名称
后面的为运行的命令


注意：运行完的容器也是占空间的，如果希望容器运行完以后立即删掉，可以使用命令`docker run --rm -it 镜像名称`

In [7]:
!docker run -it java java -version

openjdk version "1.8.0_111"
OpenJDK Runtime Environment (build 1.8.0_111-8u111-b14-2~bpo8+1-b14)
OpenJDK 64-Bit Server VM (build 25.111-b14, mixed mode)


In [ ]:
docker run -it -p 8070:8080 -v /root/software:/software --privileged=true  docker.io/centos /bin/bash

-p 映射端口8070 本机的端口 映射的容器的端口 

-v 挂载目录/root/software 本地目录 /software容器目录，在创建前容器是没有software目录的，docker 容器会自己创建

--privileged=true 关闭安全权限，否则你容器操作文件夹没有权限

--容器名称

--CMD

docker run -itd -p 9092:9092 -v /home/admin/aa1:/var/log -e deployEnv=test hub-dev.leapstack.cn/models/rc_scores_predict:v1.4 /bin/bash -lc "gunicorn --config gunicorn.conf.py --chdir /app django_rest_web.wsgi:application"

In [9]:
# #容器退出后立即删除容器Automatically remove the container when it exits

!docker run --rm -it java java -version

openjdk version "1.8.0_111"
OpenJDK Runtime Environment (build 1.8.0_111-8u111-b14-2~bpo8+1-b14)
OpenJDK 64-Bit Server VM (build 25.111-b14, mixed mode)


### 运行多个命令

## 停止容器

In [ ]:
docker stop myjava

# 打包image

1. 创建一个 Dockerfile

        FROM harbor.leapstack.cn/base/python36:predict-base-v1
        WORKDIR /app
        ENV TZ Asia/Shanghai
        RUN echo Asia/Shanghai > /etc/timezone
        RUN mkdir -p /app
        COPY . /app/

2. 打镜像
docker build -t hub-dev.leapstack.cn/models/rc_scores_predict:v1.1 .

3. 导出镜像
docker save hub-dev.leapstack.cn/models/rc_scores_predict:v1.1 -o rc_scores_predict.img

4. 导入镜像到 docker 中
docker load -i rc_scores_predict.img

docker run -itd -p 9092:9092 -v rc_scores_predict.log:/var/log/rc_scores_predict.log -e deployEnv=test hub-dev.leapstack.cn/models/rc_scores_predict:v1.1 /bin/bash -lc "gunicorn --config gunicorn.conf.py --chdir /app django_rest_web.wsgi:application"

# nb_export

In [2]:
from nbdev.export import *
notebook2script()

Converted 00_core.ipynb.
Converted 00_template.ipynb.
Converted active_learning.ipynb.
Converted algo_dl_keras.ipynb.
Converted algo_ml_eda.ipynb.
Converted algo_ml_tree_catboost.ipynb.
Converted algo_ml_tree_lgb.ipynb.
Converted algo_rs_associated_rules.ipynb.
Converted algo_rs_match_deepmatch.ipynb.
Converted algo_rs_matrix.ipynb.
Converted algo_rs_search_vector_faiss.ipynb.
Converted algo_seq_embeding.ipynb.
Converted algo_seq_features_extraction_text.ipynb.
Converted datastructure_dict_list_set.ipynb.
Converted datastructure_matrix_sparse.ipynb.
Converted engineering_concurrency.ipynb.
Converted engineering_nbdev.ipynb.
Converted engineering_panel.ipynb.
Converted engineering_snorkel.ipynb.
Converted index.ipynb.
Converted math_func_basic.ipynb.
Converted math_func_loss.ipynb.
Converted operating_system_command.ipynb.
Converted plot.ipynb.
Converted utils_functools.ipynb.
Converted utils_json.ipynb.
Converted utils_pickle.ipynb.
Converted utils_time.ipynb.


In [7]:
!nbdev_build_docs

No notebooks were modified
converting /Users/luoyonggui/PycharmProjects/nbdevlib/index.ipynb to README.md
